src/garage/torch/algos/trpo.py

```diff
@@ -72,7 +86,22 @@ def __init__(self,
                          policy_ent_coeff=policy_ent_coeff,
                          use_softplus_entropy=use_softplus_entropy,
                          stop_entropy_gradient=stop_entropy_gradient,
-                         entropy_method=entropy_method)
+                         entropy_method=entropy_method,
+                         minibatch_size=minibatch_size,
+                         max_optimization_epochs=max_optimization_epochs)
+
+    def _train(self, obs_flat, actions_flat, rewards_flat, returns_flat,
+               advs_flat):
+        policy_loss = self._train_policy(obs_flat, actions_flat, rewards_flat,
+                                         advs_flat)
+        logger.log('Policy loss: {}'.format(policy_loss))
+
+        batch_dataset = BatchDataset((obs_flat, returns_flat),
```

Comment:
why was this necessary for minibatching for the value function, but not the policy?

Reply:
I thought for TRPO, there is the `cg_iters` field for `ConjugateGradientOptimizer` and this works very similar with `max_optimization_epochs`.
Is it okay to use minibatch for ConjugateGradientOptimizer?
If so, it need to use same minibatch dataset but with different optimizing iterations? (`cg_iters` and `max_optimization_epochs`)?

Reply:
you can't use minibatching with ConjugateGradientOptimizer, or rather it would be kind of useless. Minibatching only works with SGD-based optimization algorithms, but a CG optimizer doesn't use SGD to choose parameters.

Reply:
Yes, so I used mini-batch only for value_function because default optimizer for value function is adam. When I run benchmark, I got a better result with minibatch and `optimization_epochs`

Reply:
what if i use the non-default ConjugateGradientOptimizer? will it still minibatch? wouldn't that be incorrect?

Reply:
i saw in the CHANGELOG that Torch 1.5 started making it easier to add custom optimizers, but didn't look in detail.